### PR TITLE
fix Translations subtree fetch

### PR DESCRIPTION
**Pull request by fatbeard2**
fixes https://github.com/svenfuchs/i18n-active_record/issues/79

### CODE REVIEW

**Comment by timsly** [comment]
@fatbeard2 thanks for your PR.
But can you please let me where do you need this update?
It's not very clear what is the purpose of this changes

**Comment by fatbeard2** [comment]
@timsly Sure. I tried to explain the issue here https://github.com/svenfuchs/i18n-active_record/issues/79, but I guess I was not clear enough.

Currently `I18n::Backend::ActiveRecord` and `I18n::Backend::Simple` have different behaviour when fetching translation key, that contains hash(or fetching subtree of translations). There is a test for that in `i18n` [here](https://github.com/svenfuchs/i18n/blob/master/lib/i18n/tests/lookup.rb#L17). But it covers only a case of one-level hash.  

If we have translation with a key `foo.bar.baz` in database, when I call `I18n.t 'foo'`, I expect to receive a hash of all translations under key `foo`. So basically I expect a hash like this `{ bar: { baz: 'translation' }}`. This is how `I18n::Backend::Simple` works. But  `I18n::Backend::ActiveRecord` currently returns hash like this { :"bar.baz" => 'translation'  }

I can provide more details, if needed. Let me know if you need any details.

**Comment by timsly** [comment]
Thanks for your explanation. 
We should definitely align it with Simple.
It would be also nice to add more specs for Simple and cover more complicated cases. 
I need to fix specs in the master first, I will do this in next couple of days. Then you can rebase and we can do final review.

**Comment by timsly** [comment]
@fatbeard2 master is green now.
Could you please rebase your PR on top of master?

File: lib/i18n/backend/active_record.rb [hunk 1]
```diff
@@ -56,15 +56,24 @@ def lookup(locale, key, scope = [], options = {})
...
-              hash[r.key.slice(chop_range)] = r.value
-              hash
+            result = result.inject({}) do |hash, translation|
+              hash.deep_merge build_translation_hash_by_key(key, translation)
             end
             result.deep_symbolize_keys
           end
         end
 
+        def build_translation_hash_by_key lookup_key, translation
```
Comment (timsly) [inline lib/i18n/backend/active_record.rb]: Let's wrap params here with parenthesis

**Comment by fatbeard2** [comment]
@timsly I've rebased my branch.

**Comment by timsly** [comment]
Cool. Thanks
I must admit @fatbeard2 you are quite fast 🚀

**Comment by timsly** [comment]
PR looks good 👍 
Thanks for your contribution

**Comment by fatbeard2** [comment]
@timsly Thanks! When can I expect gem version to be bumped?

**Comment by timsly** [comment]
I'll do this today

**Comment by timsly** [comment]
@fatbeard2 `0.2.0` has been released

**Pull request merged**